x-pack/plugins/security_solution/public/overview/components/detection_response/host_alerts_table/host_alerts_items.ts

```diff
@@ -0,0 +1,202 @@
+  isLoading: boolean;
+  data: AlertSeverityCounts[];
+}
+
+export const useHostAlertsItems = ({
+  skip,
+  queryId,
+}: UseHostsAlertsItemsProps): UseVulnerableHostsCountersReturnType => {
+  const { to, from, setQuery: globalSetQuery, deleteQuery } = useGlobalTime();
+  const { loading: isSignalIndexLoading, signalIndexName } = useSignalIndex();
+  const {
+    data,
+    request,
+    response,
+    setQuery,
+    loading: isLoadingData,
+    refetch: refetchQuery,
+  } = useQueryAlerts<{}, AlertCountersBySeverityAndSeverityAggregation>({
+    query: buildVulnerableHostAggregationQuery({ from, to }),
+    indexName: signalIndexName,
+    skip,
+  });
+
+  const isLoading = isLoadingData && isSignalIndexLoading;
+
+  useEffect(() => {
+    setQuery(buildVulnerableHostAggregationQuery({ from, to }));
+  }, [setQuery, from, to]);
+
+  const transformedResponse: AlertSeverityCounts[] = useMemo(() => {
+    if (data && !!data.aggregations) {
+      return pickOffCounters(data.aggregations);
+    }
+    return [];
+  }, [data]);
+
+  const refetch = useCallback(() => {
+    if (!skip && refetchQuery) {
+      refetchQuery();
+    }
+  }, [skip, refetchQuery]);
+
+  useQueryInspector({
+    deleteQuery,
+    inspect: {
+      dsl: [request],
+      response: [response],
+    },
+    refetch,
+    setQuery: globalSetQuery,
+    queryId,
+    loading: isLoading,
+  });
+  return { isLoading, data: transformedResponse };
+};
+
+export const buildVulnerableHostAggregationQuery = ({ from, to }: TimeRange) => ({
+  query: {
+    bool: {
+      filter: [
+        {
+          term: {
+            'kibana.alert.workflow_status': 'open',
+          },
+        },
+        {
+          range: {
+            '@timestamp': {
+              gte: from,
+              lte: to,
+            },
+          },
+        },
+      ],
+    },
+  },
+  size: 0,
+  aggs: {
+    [HOSTS_BY_SEVERITY_AGG]: {
+      terms: {
+        field: 'host.name',
+        order: [
+          {
+            'critical.doc_count': 'desc',
+          },
+          {
+            'high.doc_count': 'desc',
+          },
+          {
+            'medium.doc_count': 'desc',
+          },
+          {
+            'low.doc_count': 'desc',
+          },
+        ],
+        size: 4,
+      },
+      aggs: {
+        critical: {
+          filter: {
+            term: {
+              'kibana.alert.severity': 'critical',
+            },
+          },
+        },
+        high: {
+          filter: {
+            term: {
+              'kibana.alert.severity': 'high',
+            },
+          },
+        },
+        medium: {
+          filter: {
+            term: {
+              'kibana.alert.severity': 'medium',
+            },
+          },
+        },
+        low: {
+          filter: {
+            term: {
+              'kibana.alert.severity': 'low',
+            },
+          },
+        },
+      },
+    },
+  },
+});
+
+function pickOffCounters(
```

Comment:
NIT: we usually call this kind of function _parse_ something, `parseHostsData` or something similar in this case. Also, I've seen that we usually define the private functions before they are used, so maybe better move the `useHostAlertsItems` function to the bottom of the file.

Reply:
will change the name to `parse`!

Question though, and maybe I should put this to the team, but for me when I open a file I like to see the main content of the file first. Any helper /  private functions are an implementation detail that if the reader wants to see they can check out. 

So in this case, `user_host_alerts_items` is the file name and thus should be the main subject of the file, and helpers come after it. What do you think?